src/Agent.Plugins/PipelineCache/TarUtils.cs

```diff
@@ -34,8 +34,9 @@ public static class TarUtils
             {
                 File.Delete(archiveFile);
             }
+
             var processFileName = "tar";
-            var processArguments = $"-cf {archiveFile} -C {inputPath} .";
+            var processArguments = $"-cf \"{archiveFileName}\" -C \"{inputPath}\" ."; // If given the absolute path, the GNU tar fails. The workaround is to start the tarring process in the temp directory, and simply speficy 'archive.tar'
```

Comment:
I think we're going to have a problem eventually if we keeping using the same file (`archive.tar`) every time. Why not just ask for a unique temp file name (i.e. `Path.GetTempFileName()`)? It seems like we have enough "hooks" in place to properly delete this file when we're done with it. And worst case, the agent plugin host process ends prematurely and there is a large file sitting in TEMP.

Not sure it's a normal situation, but not sure what would stop someone from running 2 or more agents on the same machine, which obviously could be problematic if two builds are running at the same time.

@johnterickson  - thoughts?

Reply:
Yeah. I agree. Originally we had thought of single agent scenario and how 2 jobs won't run in parallel, and just the archive would be good, (to help us validate the manifest as well). May be Guid+archive.tar works well.

Reply:
Changed it to GUID_archive.tar

Reply:
Can we change it to %TEMP%/%GUID%/archive.tar

---
In reply to: [320374963](https://github.com/microsoft/azure-pipelines-agent/pull/2411#discussion_r320374963) [](ancestors = 320374963)